lib/class-wp-rest-blocks-search-controller.php

```diff
@@ -0,0 +1,240 @@
+<?php
+/**
+ * Start: Include for phase 2
+ * Block Areas REST API: WP_REST_Blocks_Controller class
+ *
+ * @package gutenberg
+ * @since 5.7.0
+ */
+
+/**
+ * Controller which provides REST endpoint for the blocks.
+ *
+ * @since 5.2.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Blocks_Search_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructs the controller.
+	 *
+	 * @access public
+	 */
+	public function __construct() {
+		$this->namespace = '__experimental';
+		$this->rest_base = 'blocks';
+	}
+
+	/**
+	 * Registers the necessary REST API routes.
+	 *
+	 * @access public
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
```

Comment:
A schema should be defined using `get_item_schema()` or the `schema` argument should be omitted until the schema is available. See #15981